runtime/reconcilers/canvas.go

```diff
@@ -122,3 +129,77 @@ func (r *CanvasReconciler) checkAnyComponentHasValidSpec(ctx context.Context, re
 	}
 	return false, nil
 }
+
+// validateMetricsViewTimeConsistency checks that all the metrics views referenced by the canvas' components have the same first_day_of_week and first_month_of_year.
+func (r *CanvasReconciler) validateMetricsViewTimeConsistency(ctx context.Context, refs []*runtimev1.ResourceName) error {
+	metricsViews := make(map[string]*runtimev1.Resource)
+	for _, ref := range refs {
+		// Skip non-component refs
+		if ref.Kind != runtime.ResourceKindComponent {
+			continue
+		}
+		component, err := r.C.Get(ctx, ref, false)
+		if err != nil {
+			continue
+		}
+
+		// Skip non-metrics view refs
+		for _, ref := range component.Meta.Refs {
+			if ref.Kind != runtime.ResourceKindMetricsView {
+				continue
+			}
+			// Skip if the metrics view is already in the map
+			if _, ok := metricsViews[ref.Name]; ok {
+				continue
+			}
+			// Get the metrics view
+			mv, err := r.C.Get(ctx, ref, false)
+			if err != nil {
+				if errors.Is(err, drivers.ErrResourceNotFound) {
+					return fmt.Errorf("component %q: metrics view %q not found", ref.Name, ref.Name)
```

Comment:
Maybe just `continue` here also? Since we're not trying to do wider validation, just ensure that all _valid_ referenced metrics views are consistent.